etna/models/nn/tft_native/tft.py

```diff
@@ -0,0 +1,653 @@
+        decoder_length: int,
+        n_heads: int,
+        num_layers: int,
+        dropout: float,
+        hidden_size: int,
+        lr: float,
+        static_categoricals: List,
+        static_reals: List,
+        time_varying_categoricals_encoder: List,
+        time_varying_categoricals_decoder: List,
+        time_varying_reals_encoder: List,
+        time_varying_reals_decoder: List,
+        categorical_feature_to_id: Dict,
+        loss: nn.Module,
+        optimizer_params: Optional[dict],
+    ) -> None:
+        """Init TFT.
+
+        Parameters
+        ----------
+        encoder_length:
+            encoder length
+        decoder_length:
+            decoder length
+        n_heads:
+            number of heads in Multi-Head Attention
+        num_layers:
+            number of layers in LSTM layer
+        dropout:
+            dropout rate
+        hidden_size:
+            size of the hidden state
+        lr:
+            learning rate
+        static_categoricals:
+            categorical features for the whole series, e.g. `segment`
+        static_reals:
+            continuous features for the whole series
+        time_varying_categoricals_encoder:
+            time varying categorical features for encoder
+        time_varying_categoricals_decoder:
+            time varying categorical features for decoder (known for future)
+        time_varying_reals_encoder:
+            time varying continuous features for encoder, default to `target`
+        time_varying_reals_decoder:
+            time varying continuous features for decoder (known for future), default to `target`
+        categorical_feature_to_id:
+            dictionary where keys are feature names and values are dictionaries mapping feature values to index from 0 to number of unique feature values
+        loss:
+            loss function
+        optimizer_params:
+            parameters for optimizer for Adam optimizer (api reference :py:class:`torch.optim.Adam`)
+        """
+        super().__init__()
+        self.save_hyperparameters()
+        self.encoder_length = encoder_length
+        self.decoder_length = decoder_length
+        self.n_heads = n_heads
+        self.num_layers = num_layers
+        self.dropout = dropout
+        self.hidden_size = hidden_size
+        self.lr = lr
+        self.static_categoricals = static_categoricals
+        self.static_reals = static_reals
+        self.time_varying_categoricals_encoder = time_varying_categoricals_encoder
+        self.time_varying_categoricals_decoder = time_varying_categoricals_decoder
+        self.time_varying_reals_encoder = time_varying_reals_encoder
+        self.time_varying_reals_decoder = time_varying_reals_decoder
+        self.categorical_feature_to_id = categorical_feature_to_id
+        self.loss = loss
+        self.optimizer_params = {} if optimizer_params is None else optimizer_params
+
+        self.num_embeddings_per_feature = self._num_embeddings_per_feature()
```

Comment:
Do we really need the separate method? Why can't we just place the implementation here?